yapio.c

```diff
@@ -1251,9 +1281,44 @@ yapio_alloc_buffers(const yapio_test_group_t *ytg)
         yapio_destroy_buffers_and_abort();
 }
 
+/* read input md file to initialize yapioMyRank's md array */
+static void
+yapio_initialize_source_md_buffer_from_file(const yapio_test_group_t *ytg)
+{
+    int global_sum;
+
+    /* each rank reads its own md file '{filename}-md-{rank}' */
+    char md_file[PATH_MAX + 8];
+    snprintf(md_file, PATH_MAX, "%s-md-%d", yapioTestFileName, yapioMyRank);
+
+    int nblks_per_rank = ytg->ytg_num_blks_per_rank;
+
+    FILE *file = fopen(md_file, "r");
+    int n = 0;
+
+    if (file)
+    {
+        n = 1;
+        fread(yapioSourceBlkMd, sizeof(yapio_blk_md_t), nblks_per_rank, file);
+        fclose(file);
+    }
+
+    /* make sure each rank can read its md file */
+    MPI_Allreduce(&n, &global_sum, 1, MPI_INT, MPI_SUM, MPI_COMM_WORLD);
```

Comment:
hum my bad i said something not optimal. use simple MPI_Reduce like that ; 
http://mpitutorial.com/tutorials/mpi-reduce-and-allreduce/

and have rank0 check the sum. (the sum vs nb of process)